exotica/src/KinematicTree.cpp

```diff
@@ -213,9 +215,62 @@ void KinematicTree::BuildTree(const KDL::Tree & RobotKinematics)
         }
     }
 
+
+
     setJointLimits();
 }
 
+void KinematicTree::UpdateModel()
+{
+    Root = Tree[0];
+    TreeState = Eigen::VectorXd::Zero(Tree.size());
+    for(std::shared_ptr<KinematicElement> Joint : Tree)
+    {
+        TreeMap[Joint->Segment.getName()] = Joint;
+    }
+    debugTree.resize(Tree.size()-1);
+}
+
+void KinematicTree::resetModel()
+{
+    Tree = ModelTree;
+    CollisionTree.clear();
+    UpdateModel();
+}
+
+void KinematicTree::AddElement(const std::string& name, Eigen::Affine3d& transform, const std::string& parent, shapes::ShapeConstPtr shape)
+{
+    std::shared_ptr<KinematicElement> parent_emelent;
+    if(parent=="")
+    {
+        parent_emelent = Tree[0];
+    }
+    else
+    {
+        bool found = false;
+        for(auto element : Tree)
```

Comment:
``&`` or ``const &`` missing

Reply:
I changed this but here ```auto=std::shared_ptr<KinematicElement>```. Turning it into ```const std::shared_ptr<KinematicElement>&``` does not make a lot of difference. However, it is then called with ```element->Segment.getName()``` which is valid but a bit confusing, because you are calling a reference with the ```->``` operator.